tools/c7n_gcp/c7n_gcp/client.py

```diff
@@ -177,9 +179,20 @@ def __init__(self,
         if not credentials:
             # Only share the http object when using the default credentials.
             self._use_cached_http = True
-            credentials, _ = google.auth.default(quota_project_id=project_id or
+            default_credentials, _ = google.auth.default(quota_project_id=project_id or
             get_default_project())
-        self._credentials = with_scopes_if_required(credentials, list(CLOUD_SCOPES))
+        if GOOGLE_IMPERSONATE_SERVICE_ACCOUNT:
+            impersonated_credentials = google.auth.impersonated_credentials.Credentials(
+                source_credentials=credentials or default_credentials,
+                target_principal=GOOGLE_IMPERSONATE_SERVICE_ACCOUNT,
+                target_scopes=list(CLOUD_SCOPES))
+        target_credentials = impersonated_credentials or credentials or default_credentials
+        if not impersonated_credentials:
```

Comment:
if the environment variable isn't present this fails as an undefined variable.

Reply:
Great catch, I came at this like a javascript developer so I've updated it to something more appropriate for Python.